### PR TITLE
Target -> Attention (System Rename)

### PR DESCRIPTION
**Pull request by fig02**
Note: This PR is massive. Instead of another ~10 part PR chain, I opted to do this all in one go. For reviewing convenience, each commit is reasonably bite sized chunks, many of them just find and replace operations. Please consider reviewing each commit one at a time.

Rationale: The system currently named "Target" is only responsible for only certain aspects of what one may colloquially call "Z Targeting" 
As we know, the term "Z Target" is unfortunately overloaded and not very compatible with how things are actually implemented in the code. This system rename serves to remove yet another ambiguous reference to the term "Target", and also tries to more accurately describe what it does:
```
 * The Attention System's responsibility is to bring specific actors to the player's attention.
 * This includes:
 *     - Making Navi fly over to certain actors of interest (can be lock-on actors, but may also not be)
 *     - Displaying an arrow over an actor that can be locked onto
 *     - Displaying a reticle over the current lock-on actor
 *
 * This system does not handle the actual lock-on implementation.
 * That is the responsibility of Player via `player->focusActor` and the camera.
 ```
 This inspiration for this name comes directly from the Gamecube Zelda titles (which have original name symbols). Heres a sample of some Wind Waker function names:
 ```
 dAttention_c::GetLockonList
 dAttention_c::SelectAttention
 dAttention_c::nextAttention
 dAttention_c::EnemyDistance
 dAttention_c::LockonReleaseDistanse
 ```
They even use the term lock-on too :)

### CODE REVIEW

**Comment by fig02** [comment]
For a concrete example of why this change is good (and necessary):

`ACTOR_FLAG_0` has always been thought of as the "Targetable Actor Flag". However, there are actors that Navi can fly over to but you cannot lock onto/Z target them. This flag still needs to be set for navi to see the actor and fly over to it.
`ACTOR_FLAG_TARGETABLE` doesn't work, unless you conflate "targetable" to mean "lockable or navi hoverable".

If this change is accepted, I would propose the flag to be named as `ACTOR_FLAG_ATTENTION_ENABLED`, which can be paired with (soon to be named) `ACTOR_FLAG_LOCK_ON_DISABLED` to allow Navi to fly over to it without a lock-on arrow.

File: src/overlays/actors/ovl_En_Skj/z_en_skj.c [hunk 1]
```diff
@@ -416,7 +416,7 @@ void EnSkj_Init(Actor* thisx, PlayState* play2) {
             }
 
             if ((type > 0) && (type < 3)) {
-                this->actor.targetMode = 7;
+                this->actor.attentionRangeType = 7;
```
[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_En_Skj/z_en_skj.c:419]: huh, wonder how that got passed my ctrl-fing in the previous Target Docs

**Review by krm01 — APPROVED** [review]
```c
* This system does not handle the actual lock-on implementation.
* That is the responsibility of Player via `player->focusActor` and the camera.
 ```
 this fact alone is enough to justify the change away from Target, imo. LGTM 👍

File: src/code/z_actor.c [hunk 2]
```diff
@@ -3230,30 +3232,30 @@ void Target_FindTargetableActorInCategory(PlayState* play, ActorContext* actorCt
 
     while (actor != NULL) {
         if ((actor->update != NULL) && ((Player*)actor != player) && CHECK_FLAG_ALL(actor->flags, ACTOR_FLAG_0)) {
-            // Enemy background music actor is updated here, despite not being too related to the Target system
+            // Enemy background music actor is updated here, despite not being too related to the Attention system
```
Comment (engineer124) [inline src/code/z_actor.c:3235]: Would this comment still be true? If we're interpreting this system as "Attention", then playing enemy music is a good audio cue to bring "attention" to the player that there's an enemy nearby

Comment (fig02) [reply src/code/z_actor.c:3235]: fair point yeah. even more points for this system name :D

Comment (fig02) [reply src/code/z_actor.c:3235]: Adding this to the list of responsibilities of the Attention System in z64actor.h
`- Playing enemy background music if a hostile actor is nearby`
Kind of softlaunching "hostile" as the term for `ACTOR_FLAG_2` here... soon tm

**Review by mzxrules — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]
I like this separation of the Attention system and the lock-on system. Makes a lot more sense in retrospective

**Pull request merged**